### PR TITLE
Integration based on systype

### LINKED ISSUE #657: Add ability to toggle integration tests based on machine/SYS_TYPE functionality
We need the ability to selectively add tests based on the functionality of the machine you are running the integration tests.

We should add some logic here:

https://github.com/LLNL/serac_tests/blob/8860ef692e20c9b4623db993b422f1c7522bc86a/integration/test.ats#L53

That loads a json file that describes what the current machine or SYS_TYPE is capable of running.  Initially we should toggle on CUDA.

Examples of files:

`<test_repo>/configs/blueos_3_ppc64le_ib_p9.json`
```
{"CUDA":true}
```

`<test_repo>/configs/toss_3_x86_64_ib.json`
```
{"CUDA":false}
```

`<test_repo>/configs/someones_laptop.json`
```
{"CUDA":true}
```

If we cannot match and existing config, we should emit a warning then run with these options off to try to run as many tests as possible.

Then for specific tests (like AMGx which require CUDA), we should guard adding the tests with the functionality required:

`<test_repo>/integration/some_amgx_test/some_thing.ats`
```
if SERAC_CUDA_ENABLED:
   // add tolerance test that requires AMGX
```

### PR DESCRIPTION
**Pull request by chapman39**
fixes #657

### CODE REVIEW

**Comment by chapman39** [comment]
connects with https://github.com/LLNL/serac_tests/pull/8

**Comment by chapman39** [comment]
updated doc page: https://serac.readthedocs.io/en/feature-chapman39-integration_based_on_systype/sphinx/dev_guide/testing.html

**Comment by chapman39** [comment]
AMGX tests are working properly on blueos and are skipped on quartz.

**Comment by chapman39** [comment]
Connects with https://github.com/LLNL/serac_tests/pull/8

File: ats-config/toss_3_x86_64_ib.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "CUDA": false
+}
```
Comment (white238) [inline ats-config/toss_3_x86_64_ib.json]:
```suggestion
}

```

**Review by white238 — APPROVED** [review]

**Review by jamiebramwell — APPROVED** [review]

**Pull request merged**